runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java

```diff
@@ -299,6 +299,67 @@ public Response submitEvent(@PathParam("machineId") String machineId,
+     *
+     * @param machineId the state machine identifier
+     * @return Response with execution status code
+     * @throws Exception
+     */
+    @POST
+    @Path("/{machineId}/context/eventupdate")
+    @Timed
+    public Response updateEvent(@PathParam("machineId") String machineId,
+                                EventData eventData
+    ) throws Exception {
+        try {
+            LoggingUtils.registerStateMachineIdForLogging(machineId);
+            if (eventData.getData() == null || eventData.getName() == null || eventData.getEventSource() == null) {
+                return Response.status(Response.Status.BAD_REQUEST.getStatusCode()).entity(
+                        "Event Data|Name|Source cannot be null.").build();
+            }
+            if (!(eventsDAO.findBySMIdAndName(machineId, eventData.getName()).getStatus() == Event.EventStatus.triggered)) {
+                return Response.status(Response.Status.FORBIDDEN.getStatusCode()).entity(
+                        "Input event is not in triggered state.").build();
+            }
+            boolean canUpdateEventData = false;
+            /* List holds objects retrieved from States matching input machineId and eventName as [taskId, machineId, status] */
+            List<Object[]> states = parallelScatterGatherQueryHelper.findStatesByDependentEvent(machineId, eventData.getName());
+
+            for (Object[] state : states) {
+                if (state[2] == Status.running) {
+                    return Response.status(Response.Status.NOT_ACCEPTABLE.getStatusCode()).entity(
+                            "EventData update not allowed, one of the states is in running state.").build();
+                } else if (state[2] == Status.initialized || state[2] == Status.errored || state[2] == Status.sidelined) {
+                    canUpdateEventData = true;
+                }
+            }
+            if (canUpdateEventData) {
+                this.workFlowExecutionController.persistEvent(machineId, eventData);
+
+                /* Audit entry in AuditRecord. Though audit record is in alignment with stateId, adding here
+                 * in a customised way for Event. This is temporary for audit purpose, will be adding past Event details
+                 * in a separate effort in future.
+                 * Default values: [{machineId}, 0, 0, null, null, {EventUpdateAudit} String] */
+                String EventUdpateAudit = "Event data updated for event: "+eventData.getName()+" and stateMachineId: "
+                        + machineId;
+                this.auditDAO.create(machineId, new AuditRecord(machineId, Long.valueOf(0), Long.valueOf(0),
+                        null, null, EventUdpateAudit));
+                logger.info("Event data updated for event: {} and stateMachineId: {}", eventData.getName(), machineId);
+                for (Object[] state : states) {
+                    if (state[2] == Status.errored || state[2] == Status.sidelined) {
```

Comment:
Shouldn't all the dependent states, be mark as initialized and retriggered, even if they completed succesfully,, as the eventData got updated.